tensorflow/python/keras/engine/network.py

```diff
@@ -1357,10 +1357,40 @@ def load_weights(self, filepath, by_name=False):
       if 'layer_names' not in f.attrs and 'model_weights' in f:
         f = f['model_weights']
       if by_name:
-        saving.load_weights_from_hdf5_group_by_name(f, self.layers)
```

Comment:
Don't you need to do this for save_weights as well?

Reply:
Yea. But in this I commit I did not consider about the save weights. Because usually we want to move the name based weights to object based. I will add it.